include/misc/__assert.h

```diff
@@ -86,11 +86,16 @@
 extern void posix_exit(int exit_code);
 #define __ASSERT_POST posix_exit(1)
 #else
+#if defined(CONFIG_ASSERT_ARCH_DEFINED)
+  #include "arch/cpu.h"
+  #define __ASSERT_POST __ARCH_ASSERT(__FILE__, __LINE__)
```

Comment:
Would be logical to have the posix arch use this (see 2 lines above)

Reply:
Agree